imagick/kernel_info.go

```diff
@@ -6,9 +6,145 @@ package imagick
 
 /*
 #include <magick/MagickCore.h>
+#include <magick/geometry.h>
+#include <magick/morphology.h>
 */
 import "C"
 
+//import "runtime"
+import "unsafe"
+
+// This struct represents the KernelInfo C API of ImageMagick
 type KernelInfo struct {
 	info *C.KernelInfo
 }
+
+func newKernelInfo(cki *C.KernelInfo) *KernelInfo {
+	ki := &KernelInfo{info: cki}
+	//TODO - understand and implement this.
+	// runtime.SetFinalizer(ki, Destroy)
+	//ki.IncreaseCount()
```

Comment:
This would require that you define a `Destroy()` method to satisfy the interface defined in https://github.com/gographics/imagick/blob/master/imagick/destroy.go#L7. You would have to make sure to use the correct ImageMagick free function for that type. And then relinquish the memory. It would appear you would also have to add extra increase/decrease reference count tracking support.